File: src/main/scala/resonant/lib/content/module/RenderTileDummy.java
package resonant.lib.content.module;

import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;
import universalelectricity.api.vector.Vector3;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

/** @author Calclavia */
@SideOnly(Side.CLIENT)
public class RenderTileDummy extends TileEntitySpecialRenderer
{
    @Override
    public void renderTileEntityAt(TileEntity tile, double x, double y, double z, float f)
    {
        if (tile instanceof TileBlock)
        {
            TileBlock tileBlock = ((TileBlock) tile);

            if (tileBlock.getRenderer() != null)
            {
                tileBlock.getRenderer().renderDynamic(new Vector3(x, y, z), false, f);
            }
        }
    }
}

File: src/main/scala/com/builtbroken/lib/render/wrapper/RenderTileDummy.scala
package com.builtbroken.lib.render.wrapper

import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import com.builtbroken.lib.prefab.tile.spatial.SpatialBlock
import com.builtbroken.lib.transform.vector.Vector3

/**
 * @author Calclavia
 */
@SideOnly(Side.CLIENT)
class RenderTileDummy extends TileEntitySpecialRenderer
{
  def renderTileEntityAt(tile: TileEntity, x: Double, y: Double, z: Double, f: Float)
  {
    if (tile.isInstanceOf[SpatialBlock])
    {
      val spatial = tile.asInstanceOf[SpatialBlock]
      spatial.renderDynamic(new Vector3(x, y, z), f, 0)
    }
  }
}